recommence/Checkpoint.py

```diff
@@ -58,6 +63,14 @@ def save(self) -> None:
             target=self._c.save_path,
         )
 
+    def _save_every(self) -> None:
+        def _save_every_thread(self) -> None:
+            while True:
+                self.save()
+                time.sleep(self._c.save_every_interval)
```

Comment:
I think we will want the opposite logical ordering (first wait N minutes _then_ save) since saving is expensive and there is no need to autosave immediately when the process starts